app/resources/views.py

```diff
@@ -0,0 +1,58 @@
+from flask import render_template, redirect, url_for
+from flask.ext.login import login_required, current_user
+from . import resources
+from .. import db
+from ..models import Resource, ZIPCode, Address
+from .forms import ResourceForm
+
+
+@resources.route('/', methods=['GET'])
+def index():
+    return render_template('resources/index.html')
+
+
+@resources.route('/add', methods=['GET', 'POST'])
+@login_required
+def add():
+    form = ResourceForm()
+    if form.validate_on_submit():
+        # based on form's zip code, load or create ZIPCode and add to db
+        zip_code = ZIPCode.query.filter_by(zip_code=form.zip_code.data).first()
```

Comment:
can we update this part to use the new helper methods in https://github.com/hack4impact/women-veterans-rock/blob/master/app/models/location.py/. `zip_code = ZIPCode.create_zip_code(form.zip_code.data)` might be what you're looking for, that should allow you to remove the `if zip_code is None` case.